include/roboteam_ai/stp/Play.hpp

```diff
@@ -114,9 +114,10 @@ class Play {
     rtt::ai::Field field;
 
     /**
-     * Assigns robots to roles
+     * Decides the input to the robot dealer. The result will be used to distribute the roles
+     * @return a mapping between roles and robot flags, used by the robot dealer to assign roles
      */
-    virtual void assignRoles() noexcept = 0;
+    virtual Dealer::FlagMap decideRoleFlags() noexcept = 0;
```

Comment:
(the implementations of) This function can probably have a const return type and a trailing const